plugins/processors/t128_transform/t128_transform_test.go

```diff
@@ -165,147 +165,228 @@ func TestLeavesUnmarkedFieldsInTact(t *testing.T) {
+			Result:         sample{"/rate": float64(1)},
 		},
 		{
-			Name:            "remove-mismatching-expired",
-			Fields:          map[string]string{"/rate": "/non-rate"},
-			LastSample:      map[string]interface{}{"/non-rate": 45},
-			CurrentSample:   map[string]interface{}{"/non-rate": 50},
-			TimeDelta:       6 * time.Second,
-			RemoveOriginal:  true,
-			RemainingFields: []string{},
+			Name:   "remove-mismatching-expired",
+			Fields: map[string]string{"/rate": "/non-rate"},
+			Samples: []sample{
+				{"/non-rate": 45},
+				{"/non-rate": 50},
+			},
+			TimeDelta:      6 * time.Second,
+			RemoveOriginal: true,
+			Result:         sample{},
 		},
 
 		{
-			Name:            "leave-matching-new",
-			Fields:          map[string]string{"/rate": "/rate"},
-			LastSample:      map[string]interface{}{},
-			CurrentSample:   map[string]interface{}{"/rate": 50},
-			TimeDelta:       5 * time.Second,
-			RemoveOriginal:  false,
-			RemainingFields: []string{},
+			Name:   "leave-matching-new",
+			Fields: map[string]string{"/rate": "/rate"},
+			Samples: []sample{
+				{},
+				{"/rate": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{},
+		},
+		{
+			Name:   "leave-matching-existing",
+			Fields: map[string]string{"/rate": "/rate"},
+			Samples: []sample{
+				{"/rate": 45},
+				{"/rate": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/rate": float64(1)},
+		},
+		{
+			Name:   "leave-matching-expired",
+			Fields: map[string]string{"/rate": "/rate"},
+			Samples: []sample{
+				{"/rate": 45},
+				{"/rate": 50},
+			},
+			TimeDelta:      6 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{},
+		},
+		{
+			Name:   "leave-mismatching-new",
+			Fields: map[string]string{"/rate": "/non-rate"},
+			Samples: []sample{
+				{},
+				{"/non-rate": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-rate": int64(50)},
+		},
+		{
+			Name:   "leave-mismatching-existing",
+			Fields: map[string]string{"/rate": "/non-rate"},
+			Samples: []sample{
+				{"/non-rate": 45},
+				{"/non-rate": 50},
+			},
+			TimeDelta:      5 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-rate": int64(50), "/rate": float64(1)},
 		},
 		{
-			Name:            "leave-matching-existing",
-			Fields:          map[string]string{"/rate": "/rate"},
-			LastSample:      map[string]interface{}{"/rate": 45},
-			CurrentSample:   map[string]interface{}{"/rate": 50},
-			TimeDelta:       5 * time.Second,
-			RemoveOriginal:  false,
-			RemainingFields: []string{"/rate"},
+			Name:   "leave-mismatching-expired",
+			Fields: map[string]string{"/rate": "/non-rate"},
+			Samples: []sample{
+				{"/non-rate": 45},
+				{"/non-rate": 50},
+			},
+			TimeDelta:      6 * time.Second,
+			RemoveOriginal: false,
+			Result:         sample{"/non-rate": int64(50)},
 		},
+
 		{
```

Comment:
Again, this is where the new tests start. The earlier changes are just converting to a more flexible test model. That conversion is needed because some transforms won't have a "previous" value until the third data point and the old model only allowed for two samples. The change also checks the values of the fields rather than only their names.